stdlib/public/stubs/Stubs.cpp

```diff
@@ -537,66 +535,10 @@ __swift_bool swift_stdlib_isStackAllocationSafe(__swift_size_t byteCount,
 
 __swift_bool _swift_stdlib_getCurrentStackBounds(__swift_uintptr_t *outBegin,
                                                  __swift_uintptr_t *outEnd) {
-#if defined(SWIFT_STDLIB_SINGLE_THREADED_RUNTIME)
-  // This platform does not support threads, so the API we'd call to get stack
-  // bounds (i.e. libpthread) is not going to be usable.
-  return false;
-  
-#elif defined(__APPLE__)
-  pthread_t thread = pthread_self();
-  // On Apple platforms, the stack grows down, so that the end of the stack
-  // comes before the beginning on the number line, and an address on the stack
-  // will be LESS than the start of the stack and GREATER than the end.
-  void *end = pthread_get_stackaddr_np(thread);
-  if (!end) {
-    return false;
-  }
-  *outEnd = (uintptr_t)end;
-  *outBegin = *outEnd - pthread_get_stacksize_np(thread);
-  return true;
-
-#elif defined(_WIN32) && (_WIN32_WINNT >= 0x0602)
-  ULONG_PTR lowLimit = 0;
-  ULONG_PTR highLimit = 0;
-  GetCurrentThreadStackLimits(&lowLimit, &highLimit);
-  *outBegin = lowLimit;
-  *outEnd = highLimit;
-  return true;
-
-#elif defined(__OpenBSD__)
-  stack_t sinfo;
-  if (pthread_stackseg_np(pthread_self(), &sinfo) != 0) {
+  swift::Thread::StackBounds bounds = swift::Thread::stackBounds();
```

Comment:
Nitpick, maybe have `stackBounds()` return an `llvm::Optional` or `std::optional`? I personally dislike using integer `0` as a magic value.

Reply:
Gracias. It's a nitpick to be sure—you're right that zero is not likely to be the stack's lower bound. :)